ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -413,6 +412,95 @@ def test_source_ipa_roles_check_crlmanager(self):
             assert check["kw"]["key"] == "crl_manager"
             assert check["kw"]["crlgen_enabled"] is False
 
+    def test_ipa_healthcheck_dna_plugin_returns_warning_pagure_issue_60(self):
+        """
+        This testcase checks that the status for IPADNARangeCheck on replica
+        changes from WARNING to SUCCESS when user is added on the replica
+        as the DNA range is set.
+        Issue: freeipa/freeipa-healthcheck#60
+        """
+        self.master.run_command(
+            [
+                "ipa-healthcheck",
+                "--source",
+                "ipahealthcheck.ipa.dna",
+                "--check",
+                "IPADNARangeCheck",
+                "--output-file",
+                HEALTHCHECK_OUTPUT_FILE,
+            ],
+            raiseonerr=False,
+        )
+        contents = self.master.get_file_contents(
+            HEALTHCHECK_OUTPUT_FILE, encoding="utf-8"
+        )
+        data = json.loads(contents)
```

Comment:
This needs to be factored in a function that takes as argument the json output and the check you want to check. This way you won't have duplicate code.